SocialCareCaseViewerApi/V1/Boundary/Requests/CreateAllocationRequest.cs

```diff
@@ -38,6 +38,8 @@ public CreateAllocationRequestValidator()
             RuleFor(x => x.CreatedBy)
                 .NotNull().WithMessage("Email Required")
                 .EmailAddress().WithMessage("Enter a valid email address");
+            RuleFor(x => x.AllocationStartDate)
+                .NotNull().WithMessage("Allocation start date required");
```

Comment:
I think we're missing a test like `CreateAllocationReturns400WhenInvalidCreatedBy` but for `AllocationStartDate`. 🕵️‍♀️

Reply:
I agree this test is nice, problem I have to implement is that CreateAllocationRequest datetime property of AllocationStartDate is not nullable.

This is fine because we validate it exists and it means in the rest of the code we don't require any null checks and can be certain we have access to the property AllocationStartDate.

When I try to make a test with a null AllocationStartDate the compiler will not allow it :(

Arguably because of strict typing we don't need the test but I don't like that argument in this context.